pulpcore/cli/common/generic.py

```diff
@@ -42,15 +51,58 @@ def format_help_text(
             self.help = self.help.format(entity=entity_ctx.ENTITY, entities=entity_ctx.ENTITIES)
         super().format_help_text(ctx, formatter)
 
+    def get_params(self, ctx: click.Context) -> List[click.Parameter]:
+        params = super().get_params(ctx)
+        new_params: List[click.Parameter] = []
+        # Is it more efficient to go through and just remove options from params?
+        for param in params:
+            if isinstance(param, PulpOption):
+                if param.needs_contexts is not None:
+                    if not isinstance(ctx.obj, param.needs_contexts):
```

Comment:
This will refer to the immediate context. Are you sure, we don't need to check parent contexts too? Or maybe we must not?
Also you should document that `needs_contexts` are not _all_ but only _one_ is needed.

Reply:
Should I rename it to `allowed_contexts`?

Reply:
Good call. What about `allowed_with_contexts` since the contexts are supposed to be there first.
Either is fine for me.

Reply:
Also, I think only immediate context should be check. Since parent contexts are set by the previous subcommands I don't think it should be necessary to check them since there should only be one variation of them for the current command executing, e.g. if they were different we would be executing a different command with different options.

Reply:
Yeah, maybe i was worrying too much. You still get the "nearest" object that was assigned to a context. So this can still traverse multiple subcommands if they did not introduce a new EntityContext.